src/main/java/net/mcreator/generator/GeneratorTemplate.java

```diff
@@ -101,6 +102,10 @@ public boolean shouldBeSkippedBasedOnCondition(Generator generator, Object condi
 				conditionData);
 	}
 
+	public String getPathInWorkspace(MCreator mcreator) {

```

Comment:
Since this is generator system that is technically decoupled from UI, it is better to use Workspace here than MCreator directly as an argument